test/disk_uploader_test.go

```diff
@@ -0,0 +1,150 @@
+package test
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeclientcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/utils/ptr"
+
+	kubevirtv1 "kubevirt.io/api/core/v1"
+	kubevirtcliv1 "kubevirt.io/client-go/kubecli"
+	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/constants"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/framework"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/runner"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/testconfigs"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/vm"
+)
+
+var _ = Describe("Run disk-uploader", func() {
+	f := framework.NewFramework()
+
+	BeforeEach(func() {
+		if f.TestOptions.SkipDiskUploaderTests {
+			Skip("skipDiskUploaderTests is set to true, skipping tests")
+		}
+	})
+
+	It("Extracts disk from VM and upload to container registry", func() {
+		vmName := constants.E2ETestsRandomName("example-vm")
+		dvName := constants.E2ETestsRandomName("example-dv")
+		secretName := constants.E2ETestsRandomName("disk-uploader-credentials")
+
+		config := &testconfigs.DiskUploaderTestConfig{
+			TaskRunTestConfig: testconfigs.TaskRunTestConfig{},
+			TaskData: testconfigs.DiskUploaderTaskData{
+				ExportSourceKind: "vm",
+				ExportSourceName: vmName,
+				VolumeName:       dvName,
+				ImageDestination: "quay.io/boukhano/e2e-example-vm", // TODO: Use quay.io/kubevirt/e2e-example-vm
+				SecretName:       secretName,
+			},
+		}
+		f.TestSetup(config)
+
+		createRegistryCredentials(f.CoreV1Client, f.DeployNamespace, secretName)
+		CreateAndWaitForVM(f.KubevirtClient, f.DeployNamespace, vmName, dvName, config.GetTaskRunTimeout())
+
+		runner.NewTaskRunRunner(f, config.GetTaskRun()).
+			CreateTaskRun().
+			ExpectSuccess().
+			ExpectLogs(config.GetAllExpectedLogs()...).
+			ExpectResults(nil)
```

Comment:
Query the repository after to verify disk was uploaded?

Reply:
Yep good idea, need to check it.

Reply:
Are you going to add a check?

Reply:
Yes. We may want to delete the repository after completing running the tests. If we won't delete it, and re-run the functional tests, it may be false check.

Reply:
> Are you going to add a check?

I'm not sure if it's needed. Pushing the image using go-containerregistry seems like already verifies if it was pushed successfully or not, and if not, then it's trying to retry it [1].

[1] https://github.com/google/go-containerregistry/blob/main/pkg/v1/remote/write.go#L620

Reply:
But if image already exists, the disk-uploader will repush the image and succeed.
However, these tests runs always in different namespace, and AFAIR images pushed with namespace in its name, will be treatet as a new image.

Reply:
Still, isn't the point of functional tests to verify that your task did what it says it did?

Reply:
For sure I agree, I was responding to the part about the cleanup :)

Reply:
@0xFelix I'm working on checking it, will push it soon.